ruff.schema.json

```diff
@@ -1,15 +1,15 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
```

Comment:
SchemaStore's recommendation is to still use draft-07 because many editors lack support for newer drafts, see 

> We recommend using the draft-07 JSON schema version. Later versions of JSON Schema are not yet recommended for use in SchemaStore until IDE and language support improves for those versions.

https://github.com/SchemaStore/schemastore/blob/master/CONTRIBUTING.md#best-practices

That's why I think we should not change the schema version as part of this PR